tests/Backend/Workspaces/ParallelWorkspacesTestCase.php

```diff
@@ -0,0 +1,131 @@
+<?php
+namespace Keboola\Test\Backend\Workspaces;
+
+use Keboola\Db\Import\Snowflake\Connection;
+use Keboola\StorageApi\Client;
+use Keboola\StorageApi\Options\TokenCreateOptions;
+use Keboola\StorageApi\Workspaces;
+use Keboola\Test\StorageApiTestCase;
+
+abstract class ParallelWorkspacesTestCase extends StorageApiTestCase
+{
+    /** @var string */
+    protected $runId;
+
+    /** @var Client */
+    protected $workspaceSapiClient;
+
+    public function setUp()
+    {
+        parent::setUp();
+        $this->initEmptyTestBucketsForParallelTests();
+
+        $description = get_class($this) . '\\' . $this->getName();
+
+        $this->deleteOldTestWorkspaces($description);
+
+        $this->workspaceSapiClient = new Client([
+            'token' => $this->initTestToken($description),
+            'url' => STORAGE_API_URL,
+            'backoffMaxTries' => 1,
+            'jobPollRetryDelay' => function () {
+                return 1;
+            },
+        ]);
+    }
+
+    private function deleteOldTestWorkspaces($testName)
+    {
+        $workspaces = new Workspaces($this->_client);
+
+        $oldTestWorkspaces = array_filter(
+            $workspaces->listWorkspaces(),
+            function (array $workspace) use ($testName) {
+                return $workspace['creatorToken']['description'] === $testName;
+            }
+        );
+
+        foreach ($oldTestWorkspaces as $workspace) {
+            $workspaces->deleteWorkspace($workspace['id'], [
+                'async' => true,
```

Comment:
Proč? Tím snižuješ šanci, že to nestihne smazat, než se spustí test. Resp. můžeš to takhle udělat, ale musíš za to dát while()+sleep test, že už žádné z těch workspaců neexistují. Jinak je to IMHO nestabilní.

Reply:
Jak nestihne smazat? Proč sleep? Vůbec téhle připomínce nerozumím.

Reply:
https://github.com/keboola/storage-api-php-client/pull/560#discussion_r565984514

Reply:
Jo, sorry, už! Async dělá jen to, že se to nahrne do fronty, ale ta PHP metoda vlastně čeká!